specs/lightclient/p2p-interface.md

```diff
@@ -0,0 +1,101 @@
+- _[IGNORE]_ There has been no other valid sync committee contribution for the declared `slot` by the validator referenced by `contribution_and_proof.aggregator_index`.
+- _[REJECT]_ `contribution_and_proof.selection_proof` selects the validator as an aggregator for the slot --
+  i.e. `is_sync_committee_aggregator(state, contribution.slot, contribution_and_proof.selection_proof)` returns `True`.
+- _[REJECT]_ The `contribution_and_proof.selection_proof` is a valid signature
+  of the `contribution.slot` by the validator with index `contribution_and_proof.aggregator_index`.
+- _[REJECT]_ The aggregator signature, `signed_contribution_and_proof.signature`, is valid.
+
+#### Sync committee subnets
+
+Sync committee subnets are used to propagate unaggregated sync committee signatures to subsections of the network.
+
+##### `sync_committee_{subnet_id}`
+
+The `sync_committee_{subnet_id}` topics are used to propagate unaggregated sync committee signatures to the subnet `subnet_id` to be aggregated before being gossiped to the global `sync_committee_contribution_and_proof` topic.
+
+The following validations MUST pass before forwarding the `sync_committee_signature` on the network:
+
+- _[IGNORE]_ The signature's slot is for the current slot, i.e. `sync_committee_signature.slot == current_slot`.
+- _[REJECT]_ The `signature` is valid for the message `beacon_block_root` for the validator referenced by the `index`.
+- _[REJECT]_ The `subnet_id` is correct, i.e. `subnet_id == compute_subnet_for_sync_commitee(sync_committee_signature.index)`
+- _[IGNORE]_ There has been no other valid sync committee signature for the declared `slot` by the validator referenced by `index`.
+- _[IGNORE]_ The block being signed over (`sync_committee_signature.beacon_block_root`) has been seen (via both gossip and non-gossip sources).
+
+#### Sync committees and aggregation
+
+The aggregation scheme closely follows the design of the attestation aggregation scheme. Sync committee signatures are broadcast into "subnets" defined by a topic. The number of subnets is defined by `SYNC_COMMITTEE_SUBNET_COUNT` in the [HF1 validator guide](./validator.md#constants). Sync committee members are assigned to a single subnet for the duration of their tenure in the sync committee. 
+
+Unaggregated signatures (along with metadata) are sent as `SyncCommitteeSignature`s on the `sync_committee_{subnet_id}` topics.
+
+Aggregated sync committee signatures are packaged into (signed) `SyncCommitteeContribution` along with proofs and gossiped to the `sync_committee_contribution_and_proof` topic.
+
+## The discovery domain: discv5
+
+The `attnets` key of the ENR is extended to cover both the beacon committee subnets and the sync committee subnets under a combination key `subnets`. 
+
+The `subnets` key is an SSZ `Bitvector` of length `ATTESTATION_SUBNET_COUNT + SYNC_COMMITTEE_SUBNET_COUNT`. 
```

Comment:
Currently we have a nice benefit that The ENR subnets key is a `uint64`, I'd like to keep that. Also, I think adding another ENR field isn't that bad, and much cleaner w.r.t. updating: in the double bitvector-length approach non-fork-ready nodes won't be able to parse the attnets data of fork-ready nodes.

And if we really don't want those few extra key bytes, can we at least define the attnets value as SSZ container of two bitvectors? (SSZ encodes it the exact same in this case actually).

Another reason for separation by key: if the attestation subnet count changes (it's been discussed before, can happen without shard increase even), the concatenated bitvector approach gets even messier: different node versions interpret overlapping bitfield bits differently.

Reply:
moving ahead w/ a separate bitfield for the sync committee subnets; agreed we can alter if we need more ENR space